### PR TITLE
gh-107801: Improve the accuracy of os.lseek docs

### PR DESCRIPTION
**Pull request by erlend-aasland**
- name the last parameter *whence*, like it is for seek() methods on
  file objects
- add param docstrings
- structure the valid *whence* values as a list

* Issue: gh-107801

----
:books: Documentation preview :books:: https://cpython-previews--107935.org.readthedocs.build/

### CODE REVIEW

File: Modules/posixmodule.c [hunk 1]
```diff
@@ -10422,19 +10422,24 @@ os_lockf_impl(PyObject *module, int fd, int command, Py_off_t length)
...
+        Position, interpreted relative to 'whence'.
+    whence as how: int
+        The relative position to seek from. Valid values are:
+        - SEEK_SET: seek from the start of the file.
+        - SEEK_CUR: seek from the current file position.
+        - SEEK_END: seek from the end of the file.
     /
 
-Set the position of a file descriptor.  Return the new position.
+Set the position of a file descriptor, and return the new position.
```
Comment (AA-Turner) [inline Modules/posixmodule.c]: Perhaps:
```suggestion
Set and return the position of a file descriptor.
```

File: Modules/posixmodule.c [hunk 2]
```diff
@@ -10422,19 +10422,24 @@ os_lockf_impl(PyObject *module, int fd, int command, Py_off_t length)
...
 
-Return the new cursor position in number of bytes
-relative to the beginning of the file.
+The return value is the number of bytes relative to the beginning of the file.
```
Comment (AA-Turner) [inline Modules/posixmodule.c:10439]: I'm not sure this is needed, but perhaps explaining that 'position' is measured in bytes is useful?

File: Doc/library/os.rst [hunk 3]
```diff
@@ -1163,13 +1163,16 @@ as internal buffering of data.
...
 
-.. function:: lseek(fd, pos, how, /)
+.. function:: lseek(fd, pos, whence, /)
 
    Set the current position of file descriptor *fd* to position *pos*, modified
-   by *how*: :const:`SEEK_SET` or ``0`` to set the position relative to the
-   beginning of the file; :const:`SEEK_CUR` or ``1`` to set it relative to the
-   current position; :const:`SEEK_END` or ``2`` to set it relative to the end of
-   the file. Return the new cursor position in bytes, starting from the beginning.
+   by *whence*, and return the new position in bytes, relative to
```
Comment (AA-Turner) [inline Doc/library/os.rst]:
```suggestion
   by *whence*, and return the new position in bytes relative to
```

**Review by AA-Turner — APPROVED** [review]

**Review by pitrou — APPROVED** [review]

File: Modules/posixmodule.c (hunk 1 above)
Comment (erlend-aasland) [reply Modules/posixmodule.c]: I think this is good.

File: Modules/posixmodule.c (hunk 2 above)
Comment (erlend-aasland) [reply Modules/posixmodule.c:10439]: I think it is ok, since this is a low-level API.

File: Modules/posixmodule.c (hunk 1 above)
Comment (erlend-aasland) [reply Modules/posixmodule.c]: OTOH, the suggested change is very explicit and clear. Perhaps it makes for less confusion.

Comment (AA-Turner) [reply Modules/posixmodule.c]: No strong opinion, it was an attempt to reduce some duplication -- but just a cosmetic change rather than substance.

Comment (erlend-aasland) [reply Modules/posixmodule.c]: I'll ponder on it until tomorrow :)

**Review by serhiy-storchaka — COMMENTED** [review]
See my comment in https://github.com/python/cpython/pull/107936#pullrequestreview-1578033303. These changes are not independent and should be considered in one PR.

**Comment by erlend-aasland** [comment]
> See my comment in [#107936 (review)](https://github.com/python/cpython/pull/107936#pullrequestreview-1578033303). These changes are not independent and should be considered in one PR.

I think it is fine to document constants in one PR and update the docs of the function in another PR. Documenting the constants also unlock my draft PR to clean up the Sphinx warnings in Lib/library/os.rst.

File: Modules/posixmodule.c (hunk 1 above)
Comment (erlend-aasland) [reply Modules/posixmodule.c]: I'll revert this line; the original is fine, and my suggestion neither improves nor worsens the situation.

Comment (erlend-aasland) [reply Modules/posixmodule.c]: Reverted in ca27fca6e967b2cc43d8ad465bc1354515c9fb8e

**Comment by erlend-aasland** [comment]
I'll hold this until #107936 has landed.

**Review by AA-Turner — APPROVED** [review]
Looks good!

A

**Pull request merged**